### ISSUE TITLE
TC200 false positive for annotations inside `if TYPE_CHECKING`

### ISSUE THREAD

**Issue by Daverball** [snok/flake8-type-checking#163]
Currently the following will cause a TC200 for `TypeAlias`, even though it shouldn't:

```python
from typing import TYPE_CHECKING
if TYPE_CHECKING:
    from typing_extensions import TypeAlias
    Foo: TypeAlias = str | float  # TC200 – Annotation 'TypeAlias' needs to be made a string literal
```

The same is true if you define a `Protocol`/`TypedDict`/`NamedTuple` inside an `if TYPE_CHECKING:` block if they contain an annotation for a symbol that was imported inside the block:

```python
from typing import TYPE_CHECKING
if TYPE_CHECKING:
    from collections.abc import Sequence
    from typing import NamedTuple, Protocol, TypedDict

    class FooTuple(NamedTuple):
        seq: Sequence[int]  # TC200 – Annotation 'Sequence' needs to be made a string literal

    class FooProtocol(Protocol):
        seq: Sequence[int]  # TC200 – Annotation 'Sequence' needs to be made a string literal

    class FooDict(TypedDict):
        seq: Sequence[int]  # TC200 – Annotation 'Sequence' needs to be made a string literal
```

**Comment by sondrelg** [comment]
```python
from typing import TYPE_CHECKING
if TYPE_CHECKING:
    from typing_extensions import TypeAlias
    Foo: TypeAlias = str | float  # TC200 – Annotation 'TypedDict' needs to be made a string literal
```

is confusing me a bit. Should it be this?

```python
from typing import TYPE_CHECKING
if TYPE_CHECKING:
    from typing_extensions import TypeAlias
    Foo: TypeAlias = str | float  # TC200 – Annotation 'TypeAlias' needs to be made a string literal
```

**Comment by Daverball** [comment]
Yeah that was a brain fart, since I manually typed out the error.

**Comment by sondrelg** [comment]
In general though, how I'm understanding you is that no TC200s should trigger inside a `TYPE_CHECKING` block. Have I got that right?

**Comment by Daverball** [comment]
Yes, I think that would be the most pragmatic solution, since all the other possible errors (like use before definition) should already covered by other rules.

**Comment by sondrelg** [comment]
Yeah I think that makes a lot of sense :+1: Interested in creating a PR with a fix? There's already a function that checks if you're inside a type-checking block, so should be relatively straight forward to implement :+1: Just need to add a test case or two.

**Comment by Daverball** [comment]
Sure, I'm happy to take a stab at it. I've been meaning to dabble in writing my own flake8 plugin anyways.
